ncm-mysql/src/main/pan/components/mysql/schema.pan

```diff
@@ -34,10 +34,19 @@ function component_mysql_check_db_script = {
   };
 };
 
+# The function validating password imposes a non-empty string.
+function component_mysql_password_valid = {
+  if (  match(SELF, '^[^\\]+$') ) {
+    true;
+  } else {
+    false;
+  };
+};
+
 type component_mysql_user_right = string with match(SELF, '^(ALL( PRIVILEGES)?|ALTER( ROUTINE)?|CREATE( (ROUTINE|TEMPORARY TABLES|USER|VIEW))?|DELETE|DROP|EVENT|EXECUTE|FILE|GRANT OPTION|INDEX|INSERT|LOCK TABLES|PROCESS|REFERENCES|RELOAD|REPLICATION (CLIENT|SLAVE)|SELECT|SHOW (DATABASES|VIEW)|SHUTDOWN|SUPER|TRIGGER|UPDATE|USAGE)$');
 
 type component_mysql_db_user = {
-  'password'  : string with match(SELF, '^[^\\]+$')
+  'password'  : string with (length(SELF) == 0) || component_mysql_password_valid(SELF)
```

Comment:
I really disagree on both points. MySQL allows empty passwords so this is not crippy to allow them in the configuration module... And as soon as you want to support them, there is no reason why the function should impose it and use a complex mechanism to reallow it (the global variable solution is particularly weird as you may want to enforce something different for the admin password and the user passwords.
As I said, I'd appreciate that this PR is merged in 15.4 as it really breaks backward compatibility in the sense that you cannot install it without updating your local config, if you configure some grid components with the template library and have not defined a password for some services... And I also agree that this default behaviour should be rediscussed and it would be fine with me if we open an issue to do it in 15.6.